DSpace/DSpace
# provide new PDF CoverPage implementation

Provide an new PDF coverpage generator:

* fixes #5590
* adds a template for the cover page that can be changed with an editor so that the cover page is easier to customize/theme

This PR is provided by The Library Code with support of Gesis Leibniz Institut für Sozialwissenschaften.

## Description

While cover pages can have issues with google scholar if you archive a publication in your repository that was published already somwhere else, the cover pages makes sense for repositories that publishes original contents. These repositories often wants to add a branding to there coverpages. So far, the design of the cover page was hard-coded and pretty basic in the original page generator. We decided to introduce JasperReport templates to make it possible to customize/theme the coverpages.

The cover page generator embeds metadata from the item. In the past the generator had problems with characters that are not ISO-8859-1 compatible. Our implementation fixes this problem.

To avoid to have to different implementations for the same functionality we removed the original implementation and replaced it.

## Instructions for Reviewers

List of changes in this PR:
* new dependency added: 
  * jasperreports (https://github.com/TIBCOSoftware/jasperreports) - LGPL
* Old coverpage code was removed
  * see _CitationDocumentService_ 
* New implementation was added
  * see _CoverPageService_ and changes in _CitationDocumentService_
* config for old implementation was removed
* simple JasperReports template was added _cover-template.jrxml_ 

To test the feature:
* enable feature: _citation-page.enable_globally=true_
* Download the PDF from an item (not as admin)

You should get a PDF with coverpage (compare example in attachment)

Note: The download itself does not work reliably because of https://github.com/DSpace/DSpace/issues/9665

You can use curl to download and tell it to ignore the _content-length_ header: 

```
❯ curl -http://localhost:8080/server/api/core/bitstreams/bd80ceb5-2c30-4bde-b020-467686177fe9/content --output test.pdf --ignore-content-length
```

## Checklist
_This checklist provides a reminder of what we are going to look for when reviewing your PR. You need not complete this checklist prior to creating your PR (draft PRs are always welcome).
However, reviewers may request that you complete any actions in this list if you have not done so. If you are unsure about an item in the checklist, don't hesitate to ask. We're here to help!_


## additional information
The editor for jasper report templates can be downloaded free of charge, but you have to register yourself with your email address. It's available for Linux, Windows and MacOS. See https://www.jaspersoft.com/products/jaspersoft-community for more information. Of course we checked that the license of the jasper reports library is compatible with DSpace.

See https://github.com/TIBCOSoftware/jasperreports for information about jasper reports.
